internal/config/options.go

```diff
@@ -210,19 +225,33 @@ func OptionsFromViper(configFile string) (*Options, error) {
 
 // Validate ensures the Options fields are properly formed, present, and hydrated.
 func (o *Options) Validate() error {
+	var err error
+
 	if !IsValidService(o.Services) {
 		return fmt.Errorf("%s is an invalid service type", o.Services)
 	}
 
-	// shared key must be set for all modes other than "all"
-	if o.SharedKey == "" {
-		if o.Services == "all" {
+	if IsAll(o.Services) {
+		// mutual auth between services on the same host can be generated at runtime
+		if o.SharedKey == "" {
 			o.SharedKey = cryptutil.NewBase64Key()
-		} else {
-			return errors.New("shared-key cannot be empty")
+		}
```

Comment:
Is it reasonable to be setting/hyrdating these options here in Validate or should it be done in the downstream packages?

Reply:
I _think_ doing it centrally makes sense since more than one package may use an option.  We wouldn't want different hydration/validation/default semantics by accident.